### PR TITLE
[ISSUE-901][BUG] During worker graceful shutdown, worker should report itself as unavailable and avoid master allocate slots on it.

### LINKED ISSUE #901: [BUG] Master should not allocate slots in worker is shutting down
### How to reproduce the bug?
Steps to reproduce the bug.

/cc @who-need-to-know

/assign @who-can-solve-this-bug

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related pull requests.
close #901

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX

**Comment by waitinfuture** [comment]
> /regression

https://github.com/alibaba/celeborn/actions/runs/3374496032

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala [hunk 1]
```diff
@@ -427,6 +436,10 @@ private[celeborn] class Worker(
         logInfo("Shutdown hook called.")
         shutdown.set(true)
         if (gracefulShutdown) {
+          // During graceful shutdown, to avoid allocate slots in this worker,
+          // add this worker to master's blacklist. When restart, register worker will
+          // make master remove this worker from blacklist even with save service port.
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala:441]: Remove " even with save service port".

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HAMasterMetaManager.java [hunk 2]
```diff
@@ -278,7 +278,7 @@ public void handleRegisterWorker(
   }
 
   @Override
-  public void handleReportWorkerFailure(List<WorkerInfo> failedNodes, String requestId) {
+  public void handleReportWorkerUnavailable(List<WorkerInfo> failedNodes, String requestId) {
```
Comment (waitinfuture) [inline master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HAMasterMetaManager.java:281]: Should also rename Type.ReportWorkerFailure

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala [hunk 3]
```diff
@@ -427,6 +436,10 @@ private[celeborn] class Worker(
...
+          rssHARetryClient.send(ReportWorkerUnavailable(List(workerInfo).asJava))
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala:442]: Maybe we should use askSync and ensure that master has received the msg.

File: master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HAMasterMetaManager.java (hunk 2 above)
Comment (AngersZhuuuu) [reply master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HAMasterMetaManager.java:281]: Done

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**